python/perspective/perspective/client/client.py

```diff
@@ -0,0 +1,145 @@
+        """
+        if getattr(loop, "add_future", None):
+            # Tornado IOLoop - we must manually create the future
+            self._create_future = lambda: Future()
+        elif getattr(loop, "create_future", None):
+            self._create_future = loop.create_future
+        else:
+            raise AttributeError("Loop must implement `create_future` or `add_future`!")
+        self._loop = loop
+
+    def open_table(self, name):
+        """Return a proxy Table to a `Table` hosted in a server somewhere."""
+        return PerspectiveTableProxy(self, name)
+
+    def open_view(self, name):
+        """Return a proxy View to a `View` hosted in a server somewhere."""
+        return PerspectiveViewProxy(self, name)
+
+    def _handle(self, msg):
+        """Given a response from the Perspective server, resolve the Future
+        with the response or an exception."""
+        if not msg.get("data"):
+            return
+
+        handler = self._handlers.get(msg["data"].get("id"))
+
+        if handler:
+            future = handler["future"]
+
+            keep_alive = handler.get("keep_alive")
+
+            if keep_alive and handler.get("callback_id") and future.done():
+                # Must look up callback function and execute it, and then
+                # return without re-setting the result of the Future.
+                callback = self._callback_id_cache.get(handler["callback_id"])
+                data = msg["data"]["data"]
+                if data and isinstance(data, dict):
+                    callback(**data)
+                elif data:
+                    callback(data)
+                else:
+                    callback()
+                return
+
+            if msg["data"].get("error"):
+                future.set_exception(PerspectiveError(msg["data"]["error"]))
+            else:
+                future.set_result(msg["data"]["data"])
+
+            if not keep_alive:
+                del self._handlers[msg["data"]["id"]]
+
+    def send(self, msg):
+        """Send the message to the Perspective server implementation - must be
+        implemented by a child class."""
+        raise NotImplementedError()
+
+    def post(self, msg, future=None, keep_alive=False):
+        """Given a message and an associated `Future` object, store the future
+        and send the message to the server."""
+        if self._loop is None:
+            raise PerspectiveError("An event loop must be set on `PerspectiveClient`!")
+
+        if future:
+            handler = {
+                "future": future,
+                "keep_alive": keep_alive,
+            }
+
+            if keep_alive and msg.get("callback_id"):
+                handler["callback_id"] = msg["callback_id"]
+
+            self._msg_id += 1
+            self._handlers[self._msg_id] = handler
+
+        msg["id"] = self._msg_id
+
+        self.send(msg)
+
+    def table(self, data, index=None, limit=None, name=str(random())):
```

Comment:
The default argument to `name` here is evaluated once when the method is created, so all proxy tables will have the same name.

Reply:
yup - that is a silly mistake on my part